arch/src/riscv64/layout.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright © 2024, Institute of Software, CAS. All rights reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//
+// Memory layout of riscv64 guest:
+//
+// RAM end    +---------------------------------------------------------------+
+// (dynamic,  |                                                               |
+// including  |                                                               |
+// hotplug    ~                   ~                       ~                   ~
+// memory)    |                                                               |
+//            |                            DRAM                               |
+//            ~                   ~                       ~                   ~
+//            |                                                               |
+//            |                                                               |
+// 0x80000000 +---------------------------------------------------------------+
+//            |                             ...                               |
+// 0x08100000 +---------------------------------------------------------------+
+//            |                                                               |
```

Comment:
FYI, we don't support virtio-mmio so you will need to use PCI for this

Reply:
noted, this is an unofficial draft, I will force push this branch next week :)